test/functional/test_framework/util.py

```diff
@@ -334,16 +334,25 @@ def initialize_datadir(dirname, n, chain):
     datadir = get_datadir_path(dirname, n)
     if not os.path.isdir(datadir):
         os.makedirs(datadir)
-    # Translate chain name to config name
+    write_config(os.path.join(datadir, "bitcoin.conf"), n, chain)
+    os.makedirs(os.path.join(datadir, 'stderr'), exist_ok=True)
+    os.makedirs(os.path.join(datadir, 'stdout'), exist_ok=True)
+    return datadir
+
+
+def write_config(config_path, n, chain, extra_config=""):
```

Comment:
```suggestion
def write_config(config_path, *, n, chain, extra_config=""):
```

nit: would be good to force named args to avoid mixing up `n` and `chain`. E.g. setting `chain=0` will work, but is a confusing way to say mainnet.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17556#discussion_r484036704

> nit: would be good to force named args to avoid mixing up `n` and `chain`. E.g. setting `chain=0` will work, but is a confusing way to say mainnet.

Thanks, added and updated callers.